From 609a3eb9d23772ace4c04ebc1842e6ec2e59ae21 Mon Sep 17 00:00:00 2001
From: "Damien.Horsley" <Damien.Horsley@imgtec.com>
Date: Mon, 13 Jul 2015 16:17:09 +0100
Subject: clk: pistachio: Add audio pll rates

Change the audio pll to variable and create an audio pll
rate table with two entries

Change-Id: Ifc5d925804816761f5b67ce2e3f1dbeb7447d9bf
Signed-off-by: Damien.Horsley <Damien.Horsley@imgtec.com>
---
 drivers/clk/pistachio/clk-pistachio.c | 28 ++++++++++++++++++++++++++--
 1 file changed, 26 insertions(+), 2 deletions(-)

diff --git a/drivers/clk/pistachio/clk-pistachio.c b/drivers/clk/pistachio/clk-pistachio.c
index 866b6d6..6451c6d 100644
--- a/drivers/clk/pistachio/clk-pistachio.c
+++ b/drivers/clk/pistachio/clk-pistachio.c
@@ -177,11 +177,35 @@ static struct pistachio_pll_rate_table wifi_pll_rates[] = {
 	},
 };
 
+static struct pistachio_pll_rate_table audio_pll_rates[] = {
+	{
+		.fref		= 52000000,
+		.fout_min	= 146250000,
+		.fout		= 147456000,
+		.fout_max	= 149499999,
+		.refdiv		= 0x1,
+		.fbdiv		= 0x2d,
+		.frac		= 0x5efee6,
+		.postdiv1	= 0x4,
+		.postdiv2	= 0x4,
+	}, {
+		.fref		= 52000000,
+		.fout_min	= 133250000,
+		.fout		= 135475200,
+		.fout_max	= 136500000,
+		.refdiv		= 0x1,
+		.fbdiv		= 0x29,
+		.frac		= 0xaf46fd,
+		.postdiv1	= 0x4,
+		.postdiv2	= 0x4,
+	},
+};
+
 static struct pistachio_pll pistachio_plls[] __initdata = {
 	PLL(CLK_MIPS_PLL, "mips_pll", "xtal", PLL_GF40LP_LAINT, 0x0,
 	    mips_pll_rates),
-	PLL_FIXED(CLK_AUDIO_PLL, "audio_pll", "audio_refclk_mux",
-		  PLL_GF40LP_FRAC, 0xc),
+	PLL(CLK_AUDIO_PLL, "audio_pll", "audio_refclk_mux", PLL_GF40LP_FRAC,
+	    0xc, audio_pll_rates),
 	PLL_FIXED(CLK_RPU_V_PLL, "rpu_v_pll", "xtal", PLL_GF40LP_LAINT, 0x20),
 	PLL_FIXED(CLK_RPU_L_PLL, "rpu_l_pll", "xtal", PLL_GF40LP_LAINT, 0x2c),
 	PLL_FIXED(CLK_SYS_PLL, "sys_pll", "xtal", PLL_GF40LP_FRAC, 0x38),
-- 
1.9.1

